.github/workflows/md-link-check.yml

```
@@ -18,14 +18,30 @@ concurrency:
+              - '**/*.md'
+
   lychee:
     name: Check Markdown Links
+    needs: check-for-markdown-files
     # pull_request_target fires even when GITHUB_TOKEN creates the PR (e.g. the
     # release-notes bot). pull_request does not. We register pull_request_target
     # so bot-created PRs get a "skipped" conclusion that satisfies branch
     # protection, but we skip all real work: pull_request already covers human
     # PRs, and running twice would cause the concurrency group to cancel one run.
-    if: github.event_name != 'pull_request_target'
+    if: github.event_name != 'pull_request_target' && ${{ needs.check-changes.outputs.src == 'true' }}
```

actionlint (1.7.12)

[error] if: condition "github.event_name != 'pull_request_target' && ${{ needs.check-changes.outputs.src == 'true' }}" is always evaluated to true because extra characters are around ${{ }}

(if-cond)